tests/integration/test_pr_agent_workflow_specific.py

```
@@ -0,0 +1,492 @@
+class TestPRAgentWorkflowDuplicateKeyRegression:
+    """Regression tests for the duplicate Setup Python key fix."""
+    
+    @pytest.fixture
+    def workflow_file(self) -> Path:
+        """Return path to pr-agent.yml workflow file."""
+        return Path('.github/workflows/pr-agent.yml')
+    
+    @pytest.fixture
+    def workflow_content(self, workflow_file: Path) -> Dict[str, Any]:
+        """Load and parse the workflow YAML content."""
+        with open(workflow_file, 'r', encoding='utf-8') as f:
```

Ruff (0.14.5)

Unnecessary mode argument

Remove mode argument

(UP015)